scripts/preprocess_data.py

```diff
@@ -20,6 +27,69 @@
 from mace.tools import torch_geometric
 from mace.modules import compute_statistics
 
+import concurrent.futures
+import multiprocessing as mp
+import time
+import os
+
+results = []
+
+def response(result):
+    results.append(result)
+
+    
+def target(receivers):
+    _, counts = torch.unique(receivers, return_counts=True)
+    return counts
+
+
+def neighbor_multi_process(data_loader):
+    pool = mp.Pool(processes=len(data_loader))
+    for batch in data_loader:
+        _, receivers = batch.edge_index
+        res=pool.apply_async(target, args=(receivers,), callback=response)
+
+    pool.close()
+    pool.join()
+    # breakpoint()
+    avg_num_neighbors = torch.mean(
+    torch.cat(results, dim=0).type(torch.get_default_dtype()))
+
+    return to_numpy(avg_num_neighbors).item()
+
+
+compute_stats_results = []
+
+def compute_stats_callback(result):
+    compute_stats_results.append(result)
+
+    
+def compute_stats_target(file, z_table, r_max, atomic_energies, batch_size):
+    train_dataset = data.HDF5Dataset(file, z_table=z_table, r_max=r_max)
+    train_loader = torch_geometric.dataloader.DataLoader(
+        dataset=train_dataset, 
+        batch_size=batch_size, 
+        shuffle=False,
+        drop_last=False,
+    )
+    
+    avg_num_neighbors, mean, std = compute_statistics(train_loader, atomic_energies)
+
+    output = [avg_num_neighbors, mean, std]
+    return output
+
+
+def pool_compute_stats(inputs): #inputs = (path_to_files, z_table, r_max, atomic_energies, batch_size)
+    path_to_files, z_table, r_max, atomic_energies, batch_size = inputs
+    pool = mp.Pool(processes=int(os.cpu_count()/4))
```

Comment:
This should use the `num_process` obtained from the parser

Reply:
I added the argument to the parser and it cannot find it when running 
```
python ./mace/scripts/preprocess_data.py \
    --train_file="/pscratch/sd/m/mavaylon/mace_demo/t1x_compund_subset.xyz" \
    --test_file="/pscratch/sd/m/mavaylon/mace_demo/t1x_compund_subset.xyz" \
    --num_process = 64 \
    --valid_fraction=0.05 \
    --atomic_numbers="[1, 6, 7, 8]" \
    --r_max=4.5 \
    --h5_prefix="/pscratch/sd/m/mavaylon/processed_chem/" \
    --E0s="average" \
    --seed=123 \
```